hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/EnrichmentProcessor.java

```diff
@@ -15,12 +22,20 @@
 public class EnrichmentProcessor {
 
   private static final Logger LOG = LoggerFactory.getLogger(EnrichmentProcessor.class);
-
+  private static Timer enrichmentArrivalTimer;
+  private static final String ENRICHMENT_ARRIVAL_LAG = "enrichment.arrival.lag";
   private final List<Enricher> enrichers = new ArrayList<>();
-
   public EnrichmentProcessor(List<EnricherInfo> enricherInfoList,
                              EntityDataServiceClientProvider provider) {
 
+    if (enrichmentArrivalTimer == null) {
+      synchronized (EnrichmentProcessor.class) {
+        if (enrichmentArrivalTimer == null) {
+          enrichmentArrivalTimer = PlatformMetricsRegistry
```

Comment:
I think we can create this in static block `enrichmentArrivalTimer`, we don't need to do it by acquiring lock? Any specific reason?
```
private static Timer enrichmentArrivalTimer;
static {
enrichmentArrivalTimer = PlatformMetricsRegistry.registerTimer(ENRICHMENT_ARRIVAL_LAG, new HashMap<>());
}
```

Reply:
@kotharironak  done